lib/assert.js

```diff
@@ -229,95 +230,195 @@ function assertExpectations({assertion, actual, expectations, message, prefix, s
 	}
 }
 
-function wrapAssertions(callbacks) {
-	const {pass, pending, fail} = callbacks;
-	const noop = () => {};
+const noop = () => {};
+class Assertions {
+	/**
+	 * @protected
+	 */
```

Comment:
I'm really not a fan of these kinds of comments, sorry. No need to repeat the parameters, and the `_` already indicates it's non-public.

Reply:
As you mentioned in the main comment, should we try to put this into weak map too?

Reply:
Also, the reason for the parameters in comment, is that I wanted to document what implementing method should expect as parameters passed to the given function. Is there some better way to document those values?

Reply:
TypeScript, I suppose 😉 I think the code is clear enough without the comments.